[Doc improvement][Update preview manifest to support keyboard shortcuts][3679844] (#11269)
MicrosoftDocs/msteams-docs: msteams-platform/resources/schema/manifest-schema-dev-preview.md

|`actions`| Array | | | Specifies the set of actions supported by the runtime. An action is either running a JavaScript function or opening a view such as a task pane.|
|`actions.id`| String | 64 characters | ✔️ | Specifies the ID for the action, which is passed to the code file. |
|`actions.type`| String | | ✔️ | Specifies the type of action. The `executeFunction` type runs a JavaScript function without waiting for it to finish and the `openPage` type opens a page in a given view. |

> |`actions.displayName`| String | 64 characters | | Describes the action of a [custom keyboard shortcut](/office/dev/add-ins/design/keyboard-shortcuts). This description appears in the dialog that's shown to a user when there's a shortcut conflict between multiple add-ins or with Microsoft 365. The `actions.displayName` property is required if a keyboard shortcut is defined for a specific `actions` object. Office appends the name of the add-in in parentheses after the description. For example, "Set bold (Contoso Add-in)". For more information, see [Avoid key combinations in use by other add-ins](/office/dev/add-ins/design/keyboard-shortcuts#avoid-key-combinations-in-use-by-other-add-ins). |

Reviewer: Hi @samantharamon,
1. Please remove double quotes for "Set bold" as it is not required. 
2. Change capital to small casing for all the instances of links after see. Example, 'For more information, see Avoid" to "For more information, see avoid"

> |`actions.displayName`| String | 64 characters | | Describes the action of a [custom keyboard shortcut](/office/dev/add-ins/design/keyboard-shortcuts). This description appears in the dialog that's shown to a user when there's a shortcut conflict between multiple add-ins or with Microsoft 365. The `actions.displayName` property is required if a keyboard shortcut is defined for a specific `actions` object. In addition to the description of an action, it's a good practice to also include the name of the add-in in the property, so that the user can easily identify to which add-in the action belongs. For example, "Set bold (Contoso Add-in)". For more information, see [Avoid key combinations in use by other add-ins](/office/dev/add-ins/design/keyboard-shortcuts#avoid-key-combinations-in-use-by-other-add-ins). |

Reviewer 2: My bad. Office does this automatically.

```suggestion
|`actions.displayName`| String | 64 characters | | Describes the action of a [custom keyboard shortcut](/office/dev/add-ins/design/keyboard-shortcuts). This description appears in the dialog that's shown to a user when there's a shortcut conflict between multiple add-ins or with Microsoft 365. The `actions.displayName` property is required if a keyboard shortcut is defined for a specific `actions` object. Office appends the name of the add-in in parentheses after the description. For example, "Set bold (Contoso Add-in)". For more information, see [Avoid key combinations in use by other add-ins](/office/dev/add-ins/design/keyboard-shortcuts#avoid-key-combinations-in-use-by-other-add-ins). |
```

|`alternateIcons.highResolutionIcon.size`| Number enum | | ✔️ | This property is reserved for future use. The value must be 64 (not 128). |
|`alternateIcons.highResolutionIcon.url`| String | 2048 characters | ✔️ | Specifies the full, absolute URL of the image file that is used to represent the add-in on high DPI screens. Icon image must be 128 x 128 pixels and use one of the following file formats: GIF, JPG, PNG, EXIF, BMP, TIFF.|

### extensions.keyboardShortcuts

> The `extensions.keyboardShortcuts` property defines custom keyboard shortcuts or key combinations to run specific actions. To learn more, see [Add custom keyboard shortcuts to your Office Add-ins](/office/dev/add-ins/design/keyboard-shortcuts).

Reviewer: Hi @samantharamon,
Can you please change 'To learn more' to 'For more information' to adhere to consistency and change 'see Add custom' to 'see add custom' (casing change for consistency).

| Name | Type | Maximum size | Required | Description |
|---|---|---|---|---|
|`requirements`| Object | | | Specifies the scopes, formFactors, and Office JavaScript Library requirement sets that must be supported on the Office client in order for the custom keyboard shortcuts to take effect. For more information, see [Specify Office Add-in requirements in the unified manifest for Microsoft 365](/office/dev/add-ins/develop/requirements-property-unified-manifest). |

> |`requirements.capabilities`| Array | | | Identifies the requirement sets.<br>Options: `name` (required), `minVersion`, `maxVersion`|

Reviewer: The text '<br>Options: `name` (required), `minVersion`, `maxVersion` ' can be removed as the options are listed immediately underneath.

Author: Removed.

|`requirements.capabilities.name`| String | | ✔️ | Identifies the name of the requirement set. |
|`requirements.capabilities.minVersion`| String | | | Identifies the minimum version for the requirement set. |
|`requirements.capabilities.maxVersion`| String | | | Identifies the maximum version for the requirement set. |

> |`requirements.scopes`| Array of enums | 4 | | Identifies the scopes in which the add-in can run and defines the Microsoft 365 applications in which the extension can run. <br>Supported values when specified in an `extensions.keyboardShortcuts.requirements` object: `workbook` (Excel) |

Reviewer: The text can be 'Supported value: `workbook` (Excel) as we are specifying it under the respective property.

> |`requirements.formFactors`| Array of enums | 2 | | Identifies the form factors that support the add-in.<br>Supported values when specified in an `extensions.keyboardShortcuts.requirements` object: `desktop`|

Reviewer: The text can be 'Supported value: `desktop` as we are specifying it under the respective property.

|`shortcuts`| Array | 20,000 | ✔️ | Specifies the custom keyboard shortcuts and the action each one performs. |
|`shortcuts.key`| Object | | ✔️ | Specifies custom key combinations for a single action across supported platforms. |

> |`shortcuts.key.default`| String | 32 | ✔️ | Specifies the default key combination that can be used on all supported platforms if there isn't a specific combination configured for a particular platform. A keyboard shortcut can be any combination of the characters `A-Z`, `a-z`, `0-9`, and the punctuation marks `-`, `_`, and `+`. It must include the name of at least one modifier key (Alt, Ctrl, or Shift) and only one other key. These keys must be joined with a `+` character. For example, `Ctrl+Alt+1`. The The Shift key can't be used as the only modifier key. It must be combined with either the Alt or Ctrl key. On macOS, the Alt key is mapped to the Option key. On Windows, the Command key is mapped to the Ctrl key. |

Author: Thanks for feedback, @v-ramyas5. For better discoverability, what are your thoughts on adding the link to the description of `shortcuts.key.default` instead of as a note at the bottom?

```suggestion
|`shortcuts.key.default`| String | 32 | ✔️ | Specifies the default key combination that can be used on all supported platforms if there isn't a specific combination configured for a particular platform. Keyboard shortcuts defined for an add-in must follow the [guidelines for custom key combinations](/office/dev/add-ins/design/keyboard-shortcuts#guidelines-for-custom-key-combinations).  |
```

> |`shortcuts.key.default`| String | 32 | ✔️ | Specifies the default key combination that can be used on all supported platforms if there isn't a specific combination configured for a particular platform. A keyboard shortcut can be any combination of the characters `A-Z`, `a-z`, `0-9`, and the punctuation marks `-`, `_`, and `+`. It must include the name of at least one modifier key (Alt, Ctrl, or Shift) and only one other key. These keys must be joined with a `+` character. For example, `Ctrl+Alt+1`. The The Shift key can't be used as the only modifier key. It must be combined with either the Alt or Ctrl key. On macOS, the Alt key is mapped to the Option key. On Windows, the Command key is mapped to the Ctrl key. |

Reviewer: Yes, @samantharamon, that's a good idea .

> |`shortcuts.key.default`| String | 32 | ✔️ | Specifies the default key combination that can be used on all supported platforms if there isn't a specific combination configured for a particular platform. A keyboard shortcut can be any combination of the characters `A-Z`, `a-z`, `0-9`, and the punctuation marks `-`, `_`, and `+`. It must include the name of at least one modifier key (Alt, Ctrl, or Shift) and only one other key. These keys must be joined with a `+` character. For example, `Ctrl+Alt+1`. The Shift can't be used as the only modifier key. It must be combined with either Alt or Ctrl. On macOS, the Alt key is mapped to the Option key. On Windows, the Command key is mapped to the Ctrl key. |

Reviewer: Please add the text as 'The Shift key' in this sentence and in the ones below.

|`shortcuts.key.web`| String | 32 | | Specifies the key combination of an add-in in Office on the web. A keyboard shortcut can be any combination of the characters `A-Z`, `a-z`, `0-9`, and the punctuation marks `-`, `_`, and `+`. It must include the name of at least one modifier key (Alt/Option, Ctrl/Command, or Shift) and only one other key. These keys must be joined with a `+` character. For example, `Ctrl+Alt+1`. The Shift key can't be used as the only modifier key. It must be combined with either the Alt/Option or Ctrl/Command key.|
|`shortcuts.key.windows`| String | 32 | | Specifies the key combination of an add-in in Office on Windows. A keyboard shortcut can be any combination of the characters `A-Z`, `a-z`, `0-9`, and the punctuation marks `-`, `_`, and `+`. It must include the name of at least one modifier key (Alt, Ctrl, or Shift) and only one other key. These keys must be joined with a `+` character. For example, `Ctrl+Alt+1`. The Shift key can't be used as the only modifier key. It must be combined with either the Alt or Ctrl key. |

> |`shortcuts.actionId`| String | 64 | ✔️ | Identifies the action that is performed when a keyboard shortcut is used. The `actionId` must match the `runtime.actions.id` property of an action in the `runtimes` object. |

Reviewer: Please share schema for this feature if you have it.

Author: I shared a link to the file we referenced.

|`requirements.capabilities.minVersion`| String | | | Identifies the minimum version for the requirement set. |
|`requirements.capabilities.maxVersion`| String | | | Identifies the maximum version for the requirement set. |
|`requirements.scopes`| Array of enums | 4 | | Identifies the scopes in which the add-in can run and defines the Microsoft 365 applications in which the extension can run. <br>Supported value: `workbook` (Excel) |
|`requirements.formFactors`| Array of enums | 2 | | Identifies the form factors that support the add-in.<br>Supported value: `desktop` |
|`shortcuts`| Array | 20,000 | ✔️ | Specifies the custom keyboard shortcuts and the action each one performs. |
|`shortcuts.key`| Object | | ✔️ | Specifies custom key combinations for a single action across supported platforms. |
|`shortcuts.key.default`| String | 32 | ✔️ | Specifies the default key combination that can be used on all supported platforms if there isn't a specific combination configured for a particular platform. Keyboard shortcuts defined for an add-in must follow the [guidelines for custom key combinations](/office/dev/add-ins/design/keyboard-shortcuts#guidelines-for-custom-key-combinations).  |

> |`shortcuts.key.mac`| String | 32 | | Specifies the key combination of an add-in in Office on Mac. A keyboard shortcut can be any combination of the characters `A-Z`, `a-z`, `0-9`, and the punctuation marks `-`, `_`, and `+`. It must include the name of at least one modifier key (Alt, Command, or Shift) and only one other key. These keys must be joined with a `+` character. For example, `Command+1`. The Shift key can't be used as the only modifier key. It must be combined with either the Alt or Command key. On macOS, the Alt key is mapped to the Option key. |

Reviewer: Hi @samantharamon,
Guess we can stick to the definition instead of explaining the keys supported as we have already provided the link. If we add, we must adhere to [MSTP guidelines for keys](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/term-collections/keys-keyboard-shortcuts) and the guidelines defined in keyboardshortcuts.

What are your thoughts?

Author: @v-ramyas5 That sounds good. If we just stick to the definition, would you recommend adding the link to each entry (not just `shortcuts.key.default`)? I'm leaning towards adding the link to each entry in case the reader skips directly to one of the other key properties. Let me know your thoughts on this.

Reviewer: @samantharamon, I prefer adding it as a note instead of adding of it to every field as that would be a repetitive information.

Author: Removed the guidelines from each entry and moved it to a note.
